### ISSUE TITLE
Assess scope of apostrophe bug in Analytics event reporting

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#8953]
## Description

A bug was found in the FE code that reports analytics events to Google Analytics. It was initially found to impact three Campaign Landing Pages. However, we need to explore to see if other analytics tracking might be affected by the same issue.

The bug was initially reported in the va.gov-team repo [issue 40890 in Teams repo](https://github.com/department-of-veterans-affairs/va.gov-team/issues/40890). Analysis to date (un-escaped apostrophes) is in that issue. The bug was originally discovered to affect three Campaign Landing Pages.

Note: Dave asked us to investigate whether this bug might impact other areas of the website. 

Reported impact of bug:
- No impact to site visitors/Veterans
- Events for some links on CLPs aren't recorded
- "Prevents the Medalia feedback button from firing" (per original bug report)

At least one page ([Veteran Trust in VA](https://github.com/department-of-veterans-affairs/va.gov-team/issues/40890)) was fixed by removing the apostrophe from the CTA link. It used to say "Download the VA's Trust Report": 
<img width="290" alt="image" src="https://user-images.githubusercontent.com/4054752/170762769-e1a83046-270f-4050-aeb0-374773978641.png">

## Steps / Assumptions

1. Figure out why the apostrophe isn't passing through
2. Determine if it's global 
3. Time box to roughly 2 days

NOTE: This ticket doesn't include resolution, just understanding and documenting impact in this issue.

## Acceptance Criteria
~~- [ ] Attend Analytics office hours to discuss any dependency/impact of transition to GA4 (mentioned in Team-of-teams 5/31)~~
- [ ] Figure out if there are other templates beside the CLP that use the buggy code, which fails to escape apostrophes/single-quotes (see Tim Cosgrove's [comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/40890#issuecomment-1117722080))

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Platform CMS Team`
- [ ] `Sitewide program`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by jilladams** [comment]
5 points - timeboxed

Complexity is in figuring out the global scope.

**Comment by allisonlu** [comment]
It seems like `encode` isn't an actual filter for Liquid templating. We'll have to use [the `escape` filter instead](https://shopify.dev/api/liquid/filters/string-filters#escape).

The `encode` filter also is called here: https://github.com/department-of-veterans-affairs/content-build/blob/f30b05c3db4ef402d10d13d1fedebe7bcee36508/src/site/filters/liquid.js#L827

The only other place that encode is used in the repo is in this file: https://github.com/department-of-veterans-affairs/content-build/blob/f7bafff44f67aea4cffa94ee343d31bbd8784231/src/site/paragraphs/collapsible_panel.drupal.liquid#L39

I would point it as a 3, but not entirely sure how to test the analytics (I can't replicate the screenshot in [the original ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/40890)), so testing may make it a 5.

**Comment by jilladams** [comment]
@allisonlu want to connect the dots for posterity -  
1. re: whether this bug might impact other areas of the website / if it's global, does your note mean: this is likely happening due to incorrect use of the `encode` filter, only in those 2 places? 
2. re: why the apostrophe isn't passing through - same, that's due to use of `encode` when we'll need to use `escape` instead? (In general? Or specifically for CTAs for CLPs?)

Also @wesrowe note while my brain was here: since the defect of the original report was closed, I opened https://github.com/department-of-veterans-affairs/va.gov-cms/issues/9481, which we can update as needed with outcome here. (@allisonlu if you'd rephrase anything there for whomever works on resolving this, pls feel free to edit that ticket.)
